mlx/io/gguf.cpp

```diff
@@ -81,15 +156,19 @@ std::unordered_map<std::string, array> load_gguf(
   gguf_skip_key_values_section(ctx);
   gguf_tensor tensor;
   while (gguf_get_tensor(ctx, &tensor)) {
-    std::vector<int> shape;
-    // The dimension order in GGML is the reverse of the order used in MLX.
-    for (int i = tensor.ndim - 1; i >= 0; i--) {
-      shape.push_back(tensor.dim[i]);
-    }
-    const auto& [data, dtype] = extract_tensor_data(&tensor);
-    array loaded_array = array(data, shape, dtype);
     std::string name = std::string(tensor.name, tensor.namelen);
-    result.insert({name, loaded_array});
+    const std::string weight_suffix = ".weight";
+    const size_t weight_len = weight_suffix.length();
+    const bool ends_with_weight = (name.length() > weight_len) &&
+        (name.compare(name.length() - weight_len, weight_len, weight_suffix) ==
+         0);
+    if (ends_with_weight && tensor.type == GGUF_TYPE_Q4_1) {
```

Comment:
@jbochi what i meant to say was better written here. Is there a reason you are hardcoding a check that the name is `weight` here? Should it not be sufficient that the `tensor.type === GGUF_TYPE_04_1`?

Reply:
Oh, sorry, I thought you were referring to the tensors we return, not the original tensors.

The only reason I check the suffix here is to make sure that we can correctly replace it with .scales or .biases.

We could remove the check, but in practice all quantized weights have this suffix.

Reply:
PS: note that we could support other types too. Q8_0 and Q4_0 are the ones that map more naturally without the need of re-quantization.